File: Trunk/vpPriV100Munditalia/Vatbook/Compras/EditorCompras/CmpIsEditorCompras.cs
using Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs;
using Primavera.Extensibility.Purchases.Editors;
using StdBE100;

namespace Vatbook
{
    public class CmpIsEditorCompras : EditorCompras
    {
        public override void AntesDeGravar(ref bool Cancel, ExtensibilityEventArgs e)
        {
            base.AntesDeGravar(ref Cancel, e);

            // #################################################################################################
            // ## Preenchimento automatico do CDU_Fattura_Numero para efeitos de Vatbook (JFC - 11/07/2019)
            // #################################################################################################
            if (this.DocumentoCompra.CamposUtil["CDU_Fattura_Numero"].Valor.ToString() + "" == "" | this.DocumentoCompra.CamposUtil["CDU_Fattura_Numero"].Valor.ToString() == "0")
            {
                string str;
                StdBELista lista;

                str = BSO.Compras.TabCompras.DaValorAtributo(this.DocumentoCompra.Tipodoc, "CDU_Fattura_SezionaleIVA");
                if (str + "" != "")
                {
                    lista = BSO.Consulta("select dbo.fnFattura_Num('" + str + "','c')");
                    lista.Inicio();

                    this.DocumentoCompra.CamposUtil["CDU_Fattura_Numero"] = lista.Valor(0);
                }
            }
        }
    }
}

File: Branches/VBInicial/vpPriV100GrupoMundifios/Vatbook/Compras/EditorCompras/CmpIsEditorCompras.vb
Imports Primavera.Extensibility.BusinessEntities.ExtensibilityService.EventArgs
Imports Primavera.Extensibility.Purchases.Editors
Imports Primavera.Extensibility.Attributes
Imports Vimaponto.PrimaveraV100.Clientes.GrupoMundifios.Generico
Imports StdBE100

Namespace Vatbook

    Public Class CmpIsEditorCompras
        Inherits EditorCompras

        Public Overrides Sub AntesDeGravar(ByRef Cancel As Boolean, e As ExtensibilityEventArgs)
            MyBase.AntesDeGravar(Cancel, e)

            If Module1.VerificaToken("Vatbook") = 1 Then

                '#################################################################################################
                '## Preenchimento automatico do CDU_Fattura_Numero para efeitos de Vatbook (JFC - 11/07/2019)
                '#################################################################################################
                If Me.DocumentoCompra.CamposUtil("CDU_Fattura_Numero").Valor & "" = "" Or Me.DocumentoCompra.CamposUtil("CDU_Fattura_Numero").Valor = "0" Then
                    Dim str As String
                    Dim lista As StdBELista

                    str = BSO.Compras.TabCompras.DaValorAtributo(Me.DocumentoCompra.Tipodoc, "CDU_Fattura_SezionaleIVA")
                    If str & "" <> "" Then

                        lista = BSO.Consulta("select dbo.fnFattura_Num('" & str & "','c')")
                        lista.Inicio()

                        Me.DocumentoCompra.CamposUtil("CDU_Fattura_Numero") = lista.Valor(0)

                    End If


                End If

                '#################################################################################################
                '## Preenchimento automatico do CDU_Fattura_Numero para efeitos de Vatbook (JFC - 11/07/2019)
                '#################################################################################################

            End If

        End Sub

    End Class
End Namespace